File: Daily Challenge/404. Sum of Left Leaves.cs
public class Solution 
{
    public int SumOfLeftLeaves(TreeNode root) 
    {
        return DFS(root, false);
    }

    private int DFS(TreeNode node, bool isLeft)
    {
        if (node is null) return 0;
        if (node.left is null && node.right is null && isLeft) return node.val;

        return DFS(node.left, true) + DFS(node.right, false);
    }
}

File: Daily Challenge/404. Sum of Left Leaves.py
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        return self.DFS(root, False)

    def DFS(self, node: Optional[TreeNode], isLeft: bool) -> int:
        if node is None:
            return 0

        if isLeft and node.left is None and node.right is None:
            return node.val

        return self.DFS(node.left, True) + self.DFS(node.right, False)
        
